game/src/contrib/components/XPComponent.java

```diff
@@ -3,126 +3,177 @@
 import core.Component;
 import core.Entity;
 
-import java.util.function.LongConsumer;
-
-public class XPComponent extends Component {
-
-    private static final double LEVEL_1_XP = 100;
+import java.util.function.Consumer;
+
+/**
+ * Allow associated entity to collect Experience Points (XP) and level up if enough XP is collected.
+ *
+ * <p>The component stores the current character level, the amount of XP this entity has collected,
+ * and the amount of XP this entity will drop if it dies. To drop XP, a {@link HealthComponent} is
+ * needed.
+ *
+ * <p>The callback {@link #callbackLevelUp} can be used to trigger specific behavior when the
+ * associated entity reaches a new character level.
+ *
+ * <p>Use {@link #addXP(long)} to gain XP for the component. The {@link
+ * contrib.systems.HealthSystem} will use this to increase the amount of XP in this component if the
+ * associated entity has killed another entity with an {@link XPComponent} by the amount of {@link
+ * #lootXP()} from the killed entity.
+ *
+ * <p>The {@link contrib.systems.XPSystem} will check if the component has collected enough XP to
+ * reach the next level using {@link #xpToNextCharacterLevel()} and will then update the character
+ * level and trigger the {@link #levelUp()} callback function.
+ *
+ * <p>The amount of XP needed for a level up is calculated based on the following formula: {@code
+ * FORMULA_SLOPE * (currentLevel + 1)^2 + LEVEL_1_XP}. It's a quadratic function with a slope of
+ * {@code FORMULA_SLOPE} and a y-intercept of {@code LEVEL_1_XP}.
```

Comment:
Auch hier: Der Code steht unten nochmal als Code. Hier würde eine Formel als Formel geschrieben mehr Sinn machen. Bin mir aber nicht sicher, ob das in Javadoc erlaubt ist.

Reply:
#Außerdem ist hier $(\mathrm{currLevel}+1)^2$, unten nur $(\mathrm{currLevel})^2$?

Reply:
Das sollte definitiv von außen konfigurierbar werden.

Reply:
habs konfigurierbar gemacht.
und #839